frontend/src/components/common/form/StepForm/context.tsx

```diff
@@ -119,10 +119,11 @@ export const StepperFormProvider: React.FC<IStepperFormProviderProps> = ({
   };
 
   const gotoStep = (index: number) => {
-    if (!validateCurrentStep()) {
-      return false;
-    }
     if (index >= 0 && index < steps.length) {
+      const currentIndex = getIn(values, `tabs.${values.activeTab}.activeStep`);
+      if (!validateCurrentStep() && index > currentIndex) {
```

Comment:
add an override for PA-2307 such that as long as the user is going to a previous step ignore validation failures.